### DISCUSSION TITLE
Share Shopping List and Meal Plan across Space

### DISCUSSION THREAD

**Discussion by tuetenk0pp** [TandoorRecipes/recipes#3364] [Ideas]
First of all: Thank you @vabene1111 and @smilerz for your great work over the years! I am a long time user and read every changelog with joy 😉❤️ 

I am setting up Tandoor for a small community (actually a kindergarden). We authenticate our users via Open ID Connect. As the community shares meals and cooking jobs in real life, it is important to us that every user uses the same Meal Plan and Shopping List in Tandoor as well. At the moment, this can be achieved by having the admin share it's Meal Plan and Shopping List with every individual user, BUT also, every user has to share it's Shopping List (and Meal Plan) with every other user. As you can see, this works in a group of 3, but in a group of 30 users with around half of them regularly active, this becomes a nightmare. Especially, if a user is added/signs up via SSO. For the new user to have the same experience as the old users, every one of them would have to edit their default sharing. I don't think that this is feasable.

One could approach this with additional settings on the Space level (Two checkboxes: "Shared Meal Plan", "Shared Shopping List"). Another possible solution would be the addition of user groups, to allow users to share their Meal Plan and Shopping List with a group. But I think that the introduction of the new Space settings could actually be more user friendly as it could be managed by the space admin and not every single user. And hopefully, it is easier to implement as a full user/groups system.

Of course, there is a third option: enable the feature through `.env`-vars but this would limit the flexibility as the setting would be applied globally to all Spaces.

**Comment by vabene1111** [comment]
Hi, thanks for your post and the nice feedback. 

I personally always wanted to have groups in a feature called "Households" because I think thats the easiest to explain but I also like your idea of a space setting. I wont do any database changes until the Vue 3 version reaches stability so this will definitely not happen any time soon (sorry) but it is something quite high on my priority list. I am not sure if I will do both because it introduces some complexity on the API endpoints but having a "ignore sharing and just share everything in a space" kind of feature would probably be easy. 

For now I fear you will have to either manually setup or build some kind of API automation, sorry. 

P.S. I know you have been around for a while because your avatar always reminds me of factorio for some reason even tough I think its not from there 😂

**Reply by tuetenk0pp** [reply]
> having a "ignore sharing and just share everything in a space" kind of feature would probably be easy.

This sounds great! I understand that Vue 3 is high on the priority list, no worries.

You are right, the profile picture is the rocket fuel icon from Factorio, albeit a different colour. I once had the genius plan to make a mod for the game and totally did not underestimate the undertaking. The profile picture is what is left 😅

**Reply by vabene1111** [reply]
haha so I am not mistaken, I was always wondering which fuel type it was because I did not know blue 😂

**Reply by smilerz** [reply]
Not sure exactly what your thoughts are here @vabene1111 - but the permission system probably needs an overhaul.  The Spaces were intended to be separate tenants, adding features that bleed across those tenants strikes me as adding some security risk.

A lot of the metadata is also constrained within that space as well - allowing attributes to be used across spaces is likely to lead to a fair amount of unexpected behavior.

**Reply by vabene1111** [reply]
I think I wasn't making my idea clear then: groups or households is something I plan on doing within a space not across spaces.

I don't think there should be much "inter space" features, maybe an easy way to copy recipes and possibly a combined search but I cannot think of anything else really.

**Reply by smilerz** [reply]
I probably misunderstood - I appreciate the clarification.
